src/secure_allocator.hpp

```diff
@@ -60,7 +60,7 @@ template <class T> struct secure_allocator_t
         alloc_assert (res);
         return res;
     }
-    void deallocate (T *p, std::size_t) ZMQ_NOEXCEPT { sodium_free (p); }
+    void deallocate (T *p, std::size_t) ZMQ_NOEXCEPT { if(p) sodium_free (p); }
```

Comment:
Why do you think this is necessary? It either calls `free` directly which is fine to do with a nul pointer, or it checks for nul as the first thing: https://github.com/jedisct1/libsodium/blob/master/src/libsodium/sodium/utils.c#L656

Reply:
SIGSEGV is a critical sort of severity for professional experts.
don't mention for those business customers who may never know what's C++,
make a hypothesis, those business customers have procured services/products embed libzmq;
for minimum risk and maximum assurance; 
further more, Actually we have encountered this sort of problem.

Reply:
As showed in the code above this is not really necessary with a working libsodium implementation, as it already does the exact same check as the first thing. But it doesn't hurt either and it will be a no-op, so after fixing formatting I can merged this